### PR TITLE
 Implemented record mood function

### PR DESCRIPTION
**Pull request by syedamanali77**
The function asks the user to choose a colour that matches their mood. After selecting a colour, they can add any additional comments. They also have an option to discard the entry. Once they confirm, the mood, comment, and the current timestamp get saved to their profile in the JSON file.

### CODE REVIEW

File: breeze/services/patient_service.py [hunk 1]
```diff
@@ -76,8 +81,54 @@ def edit_personal_information(self, user):
...
+                }
+
+                if hasattr(user, "add_mood_entry"):
+                    user.add_mood_entry(mood_description, comment, date_string)
+
+                self.auth_service.save_data_to_file()
+
+                print_system_message("Mood recorded successfully!")
+                direct_to_dashboard()  
+                return
```
Comment (abbieDuan87) [inline breeze/services/patient_service.py:128]: same as above

File: breeze/services/patient_service.py [hunk 2]
```diff
@@ -76,8 +81,54 @@ def edit_personal_information(self, user):
...
+        while True:
+            colour_input = input("Enter the colour code that represents your mood today: ").strip().lower()
+            if colour_input in colour_to_mood:
+                mood_description = colour_to_mood[colour_input]
+                comment = input("Would you like to add any additional comments about your mood today? Type [R] to discard your entry: ").strip()
+                
+                if comment.lower() == 'r':
+                    print_system_message("Mood entry discarded. Returning to dashboard.")
+                    direct_to_dashboard()
+                    return
```
Comment (abbieDuan87) [inline breeze/services/patient_service.py:112]: with direct_to_dashboard(), you don't need another return here

**Review by abbieDuan87 — CHANGES_REQUESTED** [review]
Lgtm just nit picks.
This branch are 2 commits behind the main, causing diverged branches so I can't pull the latest changes of this branch to test out the function on my local. You may want to pull the latest from main and rebase and push again with any new changes

File: breeze/services/patient_service.py (hunk 1 above)
Comment (syedamanali77) [reply breeze/services/patient_service.py:134]: return prevents the function from going back into the while loop. Without it, the function would keep asking the user to enter a mood, even though their entry has already been saved.

File: breeze/services/patient_service.py (hunk 2 above)
Comment (syedamanali77) [reply breeze/services/patient_service.py:118]: Without this return, the function would continue to execute the code below, like creating a mood entry even though the user wants to discard

**Review by abbieDuan87 — DISMISSED** [review]
good work!

**Comment by ben-jthompson** [comment]
Just one thing, 'case 'r':' is appearing twice within the patient dashboard navigation: 
image

**Comment by abbieDuan87** [comment]
> Just one thing, 'case 'r':' is appearing twice within the patient dashboard navigation. Works well otherwise. image

great spot

**Comment by abbieDuan87** [comment]
@syedamanali77 after you delete the duplicated part you can safely merge

**Review by ben-jthompson — APPROVED** [review]

**Pull request merged**